http/test/browser/src/common.ts

```diff
@@ -473,24 +474,24 @@ export function run(uri: string) {
 
       const ignoredRequest$ = Rx.Observable.of(uri + '/json');
       const request$ = Rx.Observable.of(uri + '/hello').delay(10);
-      const scopedRequest$ = sources.HTTP
-        .isolateSink(sources.HTTP.isolateSink(request$, 'foo'), 'something')
+      const fooInsideBarRequest$ = sources.HTTP
+        .isolateSink(sources.HTTP.isolateSink(request$, 'foo'), 'bar')
         .shareReplay();
-      const twiceScopedHTTPSource = sources.HTTP.isolateSource(
-        sources.HTTP.isolateSource(sources.HTTP, 'foo'),
-        'something',
+      const fooInsideBarHTTPSource = sources.HTTP.isolateSource(
+        sources.HTTP.isolateSource(sources.HTTP, 'bar'),
+        'foo',
```

Comment:
@jvanbruegge notice that I fixed this test. It's easy to get confused, so I remind myself that

`isolateSource :: OuterSource -> InnerSource`
`isolateSink :: InnerSink -> OuterSink`

Reply:
Your test would not fail on the original issue thou, the isolation was wrong because there was `foo` *somewhere* in the namespace, even of the deeper levels didnt match

So
```
isolationA: foo -> bar
isolationB: something -> foo

isolationB receives stuff from isolationA
```

Reply:
Did you see the new test I added? When I revert `unshift` to `push`, both this test and the new test fail.

Reply:
Do the test fail if you change it to `namespace.indexOf(scope)`?

Reply:
With `namespace.indexOf(scope)`, this test would (correctly) fail on this part:

```js
      fooInsideFooHTTPSource.select().subscribe(function(response$) {
        assert(false);
        done('should not be called');
      });
```

Reply:
:+1: